documents/xaml-suggested-actions/xaml-designer-suggested-actions-extensibility.md

```diff
@@ -0,0 +1,110 @@
+            new PropertyAction(ButtonActionProvider.Token_Property_Background, "Background"),
+            new PropertyAction(ButtonActionProvider.Token_Property_IsEnabled, "IsEnabled")
+            )); ;
+        this.AddGroup(new ActionGroup(ButtonActionProvider.Token_Group_Specific,
+            new PropertyAction(ButtonActionProvider.Token_Property_IsCancel, "IsCancel"),
+            new PropertyAction(ButtonActionProvider.Token_Property_IsDefault, "IsDefault")));
+        this.AddGroup(new BorderBrushActionGroup());
+        this.AddGroup(new VisibilityActionGroup());
+        this.AddGroup(new FontSettingsActionGroup());
+    }
+}
+```
+### Example for `MyButtonSuggestedActionProvider : ButtonActionProvider`
+```CS
+public class MyButtonSuggestedActionProvider : ButtonActionProvider
+{
+    public static ActionToken Token_Property_CustomProp = ButtonActionProvider.Token_Last + 1;
+    public new static ActionToken Token_Last = new ActionToken(0x2FFF);
+    
+    public override string Type { get => "CustomControlLibrary.MyButton"; }
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        //Hide Visibility Group
+        this.GetGroupByToken(SuggestedActionProviderTokens.Token_Group_VisibilitySettings).IsVisible = false;
+
+        //Hide IsDefault Property
+        this.GetActionByToken(MyButtonSuggestedActionProvider.Token_Property_IsDefault).IsVisible = false;
+
+        //Add new Opacity (without Token) property after IsDefault (even if it was hidden before)
+        this.InsertAction(new PropertyAction("Opacity"), after: MyButtonSuggestedActionProvider.Token_Property_IsDefault);
+
+        //Add new Link Action after IsDefault - will be inserted after IsDefault, but before Opacity
+        this.InsertAction(new LinkAction("New link", () => { }), after: MyButtonSuggestedActionProviderToken_Property_IsDefault);
+        
+        //Subscribe to ModelItem Property Changed Event to update view if needed
+        this.ModelItemPropertyChanged += MyButtonSuggestedActionProvider_ModelItemPropertyChanged;
+    }
+    
+    private void MyButtonSuggestedActionProvider_ModelItemPropertyChanged(object sender, System.ComponentModelPropertyChangedEventArgs e)
+    {
+    }
+}
+```
+
+### Metadata registration
+```CS
+...
+//Add SuggestionsAttribute to enabled "XAML Suggested Actions for control"
+builder.AddCustomAttributes("System.Windows.Controls.Button", new SuggestionsAttribute());
+
+//Add one provider
+builder.AddCustomAttributes("System.Windows.Controls.Button", new FeatureAttribute(typeof(ButtonActionProvider)));
+
+//Same for another control
+builder.AddCustomAttributes("CustomControlLibrary.MyButton", new SuggestionsAttribute());
+builder.AddCustomAttributes("CustomControlLibrary.MyButton", new FeatureAttribute(typeof(MyButtonSuggestedActionProvider)));
+...
+```
+
+### Documentation link
+At the top of 'XAML Suggested Actions' there is a Type Name label.
```

Comment:
"At the top of 'XAML Suggested Actions'" should be something like "At the top of the Suggested Actions dialog in the designer". Maybe including an image here would help clarify.